test/SlowTests/Issues/RavenDB-21150.cs

```diff
@@ -0,0 +1,105 @@
+        {
+            _testOutputHelper = testOutputHelper;
+        }
+
+        [RavenFact(RavenTestCategory.Cluster | RavenTestCategory.Replication)]
+        public async Task DontDeleteNonReplicatedTombstone()
+        {
+            var (nodes, leader) = await CreateRaftCluster(3);
+            var databaseName = GetDatabaseName();
+            var nodeToRemove = leader.WebUrl == nodes[2].WebUrl ? nodes[0] : nodes[2];
+            using (var store1 = GetDocumentStore(new Options
+            {
+                   ModifyDatabaseName = s => databaseName,
+                   ReplicationFactor = 3,
+                   Server = leader,
+                   RunInMemory = false,
+                   ModifyDocumentStore = (store) => store.Conventions.DisableTopologyUpdates = true
+            }))
+            using (var store2 = new DocumentStore
+            {
+                   Database = databaseName,
+                   Urls = new []{ nodes.First(x => x.WebUrl != leader.WebUrl && x.WebUrl != nodeToRemove.WebUrl).WebUrl},
+                   Conventions = new DocumentConventions()
+                   {
+                       DisableTopologyUpdates = true
+                   }
+            }.Initialize())
+            {
+                using (var session = store1.OpenAsyncSession())
+                {
+                    session.Advanced.WaitForReplicationAfterSaveChanges(TimeSpan.FromSeconds(30), replicas: 2);
+                    await session.StoreAsync(new User { Name = "Toli" }, "users/1");
+                    await session.SaveChangesAsync();
+                }
+
+                await DisposeServerAndWaitForFinishOfDisposalAsync(nodeToRemove);
+
+                using (var session = store1.OpenAsyncSession())
+                {
+                    session.Delete( "users/1");
+                    await session.SaveChangesAsync();
+                }
+
+                Assert.True(WaitForDocumentDeletion(store1, "users/1", 15000));
+                Assert.True(WaitForDocumentDeletion(store2, "users/1", 15000));
+                DocumentDatabase documentDatabase;
+                nodes.Remove(nodeToRemove);
+
+                foreach (var node in nodes)
+                {
+                    documentDatabase = await node.ServerStore.DatabasesLandlord.TryGetOrCreateResourceStore(databaseName);
+                    using (documentDatabase.DocumentsStorage.ContextPool.AllocateOperationContext(out DocumentsOperationContext ctx))
+                    using (ctx.OpenReadTransaction())
+                    {
+                        Assert.Equal(1, documentDatabase.DocumentsStorage.GetNumberOfTombstones(ctx));
+                    }
+                }
+
+                foreach (var node in nodes)
+                {
+                    await node.ServerStore.DatabasesLandlord.RestartDatabase(databaseName);
+                }
+
+                await Task.Delay(TimeSpan.FromSeconds(10));
```

Comment:
Let's wait here for the replication task to start working

Reply:
done